examples/ft-refund/solidity-contract/src/FtRefund.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: CC-BY-1.0
+pragma solidity ^0.8.17;
+
+// See `docs/NearFromAurora.md` for more information on what is going on in this example.
+
+import "openzeppelin-contracts/access/AccessControl.sol";
+import "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/utils/Strings.sol";
+import "./IEvmErc20.sol";
+import {AuroraSdk, Codec, NEAR, PromiseCreateArgs, PromiseResult, PromiseResultStatus, PromiseWithCallback} from "aurora-sdk/AuroraSdk.sol";
+
+uint64 constant FT_TRANSFER_CALL_NEAR_GAS = 35_000_000_000_000;
+
+uint64 constant CALLBACK_NEAR_GAS = 130_000_000_000_000;
+
+uint64 constant REFUND_NEAR_GAS = 35_000_000_000_000;
+
+uint64 constant APPROVE_NEAR_GAS = 20_000_000_000_000;
+
+contract FtRefund is AccessControl {
+    using AuroraSdk for NEAR;
+    using AuroraSdk for PromiseCreateArgs;
+    using AuroraSdk for PromiseWithCallback;
+    using Codec for bytes;
+
+    bytes32 public constant CALLBACK_ROLE = keccak256("CALLBACK_ROLE");
+    bytes16 private constant _SYMBOLS = "0123456789abcdef";
+
+    IERC20 public wNEAR;
+    string public nearAccountId;
+    NEAR public near;
+
+    constructor(string memory _nearAccountId, IERC20 _wNEAR) {
+        nearAccountId = _nearAccountId;
+        near = AuroraSdk.initNear(_wNEAR);
+        wNEAR = _wNEAR;
```

Comment:
Why we save wNEAR as well and not just use `near.wNEAR`?

Reply:
The fields of the `NEAR` struct are not public, so cannot be used outside the SDK itself. Being the maintainers of the SDK that is something we have the power to change, but I'm not sure if we want to. Exposing the fields publicly leaks some of the implementation details and we may want to change them in the future.